src/common/color/helper.ts

```diff
@@ -82,3 +87,103 @@ export const getColor = (props: Partial<ColorHelperProps>) => {
     return colorScheme;
   }
 };
+
+const getValueScale = (
+  data,
+  colorScheme: ColorSchemeType,
+  emptyColor: string
+): ColorSchemeValueScale => {
+  const valueDomain = extent(
+    uniqueBy(
+      data,
+      (d) => d.data,
+      (d) => d.value
+    )
+  );
+
+  return (point) => {
+    // For 0 values, lets show a placeholder fill
+    if (point === undefined || point === null) {
+      return emptyColor;
+    }
+
+    // Note: this can return css style values, not just colors
+    return getColor({
+      scale: scaleQuantile,
+      domain: valueDomain,
+      key: point,
+      colorScheme
+    });
+  };
+};
+
+export const getColorSchemeStyles = (
+  point,
+  valueScales: Map<string, ColorSchemeValueScale>
+): Partial<CSSStyleDeclaration> =>
+  Array.from(valueScales).reduce((acc, [key, valueScale]) => {
+    return { ...acc, [key]: valueScale(point) };
+  }, {});
+
+const getColorSchemeForProperty = (
+  colorScheme: ColorSchemeType,
+  colorSchemeProperty: string
+): ColorSchemeType => {
+  if (!Array.isArray(colorScheme)) {
+    return colorScheme;
+  }
+
+  const newColorScheme = colorScheme.map(
+    (schemeItem: string | Partial<CSSStyleDeclaration>) => {
+      if (typeof schemeItem === 'object') {
+        return schemeItem?.[colorSchemeProperty];
+      }
+
+      return schemeItem;
+    }
+  );
+  return newColorScheme;
+};
+
+/**
+ * This function creates a map of value scales for different properties based on a provided color scheme.
+ * Each value scale is a function that takes a data point and returns a css style value based on that point.
+ *
+ * @param {Array} data - The data used to create the scales.
+ * @param {ColorSchemeType} colorScheme - The color scheme used to generate the scales. This can be an array of colors or an array of objects where each object contains a set of css styles.
+ * @param {string} emptyColor - The color used for data points with no value.
+ *
+ * @returns {Map<string, ColorSchemeValueScale>} A map where each key is a property name and each value is a function that takes a data point and returns a value for the property.
+ *
+ * If the color scheme is an array of strings, they will be treated as fill values.
+ *
+ */
+export const createColorSchemeValueScales = (
+  data,
+  colorScheme: ColorSchemeType,
+  emptyColor: string
+): Map<string, ColorSchemeValueScale> => {
+  const valueScales = new Map<string, ColorSchemeValueScale>();
+
+  valueScales.set('fill', getValueScale(data, colorScheme, emptyColor));
```

Comment:
Should fill be specific call out here?

Reply:
This is in the case the color scheme is an array of color strings, they get applied to the fill. I updated this func to make it more clear